### PR TITLE
Asking the user for feedback using custom heuristic

### LINKED ISSUE #3157: Feedback given flag should be in the '.azure' config not '.azure-shell'
### Description
**Outline the issue here:**

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here:

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here:

**OS Version:** What OS and version are you using?  
Answer here:

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here:

### PR DESCRIPTION
**Pull request by oakeyc**
fixes #3157

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by oakeyc** [comment]
screen shot 2017-05-02 at 2 45 39 pm

File: src/command_modules/azure-cli-shell/azclishell/app.py [hunk 1]
```diff
@@ -521,10 +533,15 @@ def handle_scoping_input(self, continue_flag, cmd, text):
         return continue_flag, cmd
 
     def cli_execute(self, cmd):
+        """ runs a command through the CLI """
         try:
             args = parse_quotes(cmd)
             azlogging.configure_logging(args)
 
+            if 'feedback' in args:
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-shell/azclishell/app.py]: What if I create a VM with name 'feedback'.
How does this work?

Comment (oakeyc) [inline src/command_modules/azure-cli-shell/azclishell/app.py]: Changed

File: src/command_modules/azure-cli-shell/azclishell/frequency_heuristic.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def update_frequency():
+    """ updates the frequency from files """
+    with open(os.path.join(shell_config(), SHELL_CONFIG.get_frequency()), 'r') as freq:
+        try:
+            frequency = json.load(freq)
+        except ValueError:
+            frequency = {}
+
+    with open(os.path.join(shell_config(), SHELL_CONFIG.get_frequency()), 'w') as freq:
+        now = datetime.datetime.now()
```
Comment (troydai) [inline src/command_modules/azure-cli-shell/azclishell/frequency_heuristic.py]: UTC?
\

**Comment by troydai** [comment]
Looks good. Please test it again before merge.

**Pull request merged**